LFS64 interfaces are now generally considered deprecated, and are no longer
available in MUSL since version 1.2.4.

--- a/configure.ac
+++ b/configure.ac
@@ -12,6 +12,8 @@
 AC_PREREQ([2.59])
 AC_PROG_GREP
 
+AC_SYS_LARGEFILE
+
 m4_ifdef(
     [AM_SILENT_RULES],
     [AM_SILENT_RULES([yes])])
--- a/src/zfile.c
+++ b/src/zfile.c
@@ -4,7 +4,7 @@
 #include <sys/types.h>
 
 #ifdef __CYGWIN__
-typedef _off64_t off64_t;
+typedef _off64_t off_t;
 #endif
 
 #include <assert.h>
@@ -331,14 +331,14 @@
 }
 
 static int
-zfile_seek(void *cookie_, off64_t *offset_, int whence) {
+zfile_seek(void *cookie_, off_t *offset_, int whence) {
     struct zfile *cookie = cookie_;
-    off64_t new_offset = 0, offset = *offset_;
+    off_t new_offset = 0, offset = *offset_;
 
     if (whence == SEEK_SET) {
         new_offset = offset;
     } else if (whence == SEEK_CUR) {
-        new_offset = (off64_t)cookie->logic_offset + offset;
+        new_offset = (off_t)cookie->logic_offset + offset;
     } else {
         /* SEEK_END not ok */
         return -1;
@@ -348,7 +348,7 @@
         return -1;
 
     /* Backward seeks to anywhere but 0 are not ok */
-    if (new_offset < (off64_t)cookie->logic_offset && new_offset != 0) {
+    if (new_offset < (off_t)cookie->logic_offset && new_offset != 0) {
         return -1;
     }
 
